delfin/coordination.py

```diff
@@ -112,6 +119,79 @@ def get_lock(self, name):
 LOCK_COORDINATOR = Coordinator(prefix='delfin-')
 
 
+class LeaderElectionCoordinator(Coordinator):
+
+    def __init__(self, agent_id=None):
+        super(LeaderElectionCoordinator, self). \
+            __init__(agent_id=agent_id, prefix="leader_election")
+        self.group = None
+
+    def start(self):
+        """Connect to coordination back end."""
+        if self.started:
+            return
+
+        # NOTE(gouthamr): Tooz expects member_id as a byte string.
+        member_id = (self.prefix + "-" + self.agent_id).encode('ascii')
+        LOG.info('Started Coordinator (Agent ID: %(agent)s, '
+                 'prefix: %(prefix)s)', {'agent': self.agent_id,
+                                         'prefix': self.prefix})
+
+        backend_url = _get_redis_backend_url()
+        self.coordinator = coordination.get_coordinator(
+            backend_url, member_id,
+            timeout=CONF.coordination.lease_timeout)
+        self.coordinator.start()
+        self.started = True
+
+    def ensure_group(self, group):
+        # request groups
+        req = self.coordinator.get_groups()
+        groups = req.get()
+        try:
+            # check if group exist
+            groups.index(group)
+        except Exception:
+            # create a group if not exist
```

Comment:
can we put a debug message here? or for that sake there should be a debug message in any exception block, IMO

Reply:
added debug message. Thanks

Reply:
Comment starts with uppercase letter

Reply:
updated comment as per suggestion